src/formulas/compiler.ts

```diff
@@ -64,10 +64,10 @@ export function compileTokens(tokens: Token[]): CompiledFormula {
     const scope = new Scope();
 
     if (ast.type === "BIN_OPERATION" && ast.value === ":") {
-      throw new BadExpressionError(_t("Invalid formula"));
+      throw { value: CellErrorType.BadExpression, message: _t("Invalid formula") };
```

Comment:
Personnally I prefere the old way to throw/return error. Why did you change this ?

Reply:
the goal would be that the formulas end up returning errors in the form of the normal payload rather than throwing errors, (performance reason)

Reply:
I must admit I agree with Anthony. It feels very here where we are not in the context of a function implementation that would return a value (or an error).

Reply:
`Bad Expression Error(message)` and other errors could indeed result in `{value: "#SomeError", message}`